R/git.R

```diff
@@ -323,85 +323,117 @@ git_clean <- function() {
 #' Git/GitHub sitrep
 #'
 #' Get a situation report on your current Git/GitHub status. Useful for
-#' diagnosing problems.
+#' diagnosing problems. The default is to report all values; provide values
+#' for `tool` or `scope` to be more specific.
+#'
+#' @param tool which tool(s) to report: one or more of `"git"`, `"github"`.
+#' @param scope which scope(s) to report: one or more of `"global"`, `"project`.
+#'
 #' @export
 #' @examples
 #' \dontrun{
+#' # report all
 #' git_sitrep()
+#'
+#' # report git global
+#' git_sitrep("git", "global")
 #' }
-git_sitrep <- function() {
+git_sitrep <- function(tool = c("git", "github"),
+                       scope = c("global", "project")) {
+
+  tool <- rlang::arg_match(tool, multiple = TRUE)
+  scope <- rlang::arg_match(scope, multiple = TRUE)
+
   ui_silence(try(proj_get(), silent = TRUE))
 
   # git (global / user) --------------------------------------------------------
-  hd_line("Git config (global)")
-  kv_line("Name", git_cfg_get("user.name", "global"))
-  kv_line("Email", git_cfg_get("user.email", "global"))
-  kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
-  vaccinated <- git_vaccinated()
-  kv_line("Vaccinated", vaccinated)
-  if (!vaccinated) {
-    ui_info("See {ui_code('?git_vaccinate')} to learn more")
-  }
-  kv_line("Default Git protocol", git_protocol())
   init_default_branch <- git_cfg_get("init.defaultBranch", where = "global")
-  kv_line("Default initial branch name", init_default_branch)
+  if ("git" %in% tool && "global" %in% scope) {
+    cli::cli_h3("Git (global)")
+    kv_line("Name", git_cfg_get("user.name", "global"))
+    kv_line("Email", git_cfg_get("user.email", "global"))
+    kv_line("Global (user-level) gitignore file", git_ignore_path("user"))
+    vaccinated <- git_vaccinated()
+    kv_line("Vaccinated", vaccinated)
+    if (!vaccinated) {
+      ui_info("See {ui_code('?git_vaccinate')} to learn more")
+    }
+    kv_line("Default Git protocol", git_protocol())
+    kv_line("Default initial branch name", init_default_branch)
+  }
 
   # github (global / user) -----------------------------------------------------
-  hd_line("GitHub")
   default_gh_host <- get_hosturl(default_api_url())
-  kv_line("Default GitHub host", default_gh_host)
-  pat_sitrep(default_gh_host)
+  if ("github" %in% tool && "global" %in% scope) {
+    cli::cli_h3("GitHub (global)")
```

Comment:
Another place to ponder `"global"` versus `"user"`.

Reply:
proposal in https://github.com/r-lib/usethis/commit/9fd633aecff586d657008df4e95e75981d9e1e75